compositions/NFTPage/NFTPageHero.tsx

```diff
@@ -1,33 +1,53 @@
 import { ImageWithNounFallback } from 'components'
 
+import { useToken } from 'hooks/useToken'
+
+import { TypeSafeDao } from 'validators/dao'
+import { TypeSafeToken } from 'validators/token'
+
 import { cardImageWrapper } from '@media/NftMedia.css'
 import { useOptionalImageURIDecode } from '@media/hooks/useImageURIDecode'
-import { useNFTProvider } from '@shared/providers/NFTProvider'
 import { Box, BoxProps } from '@zoralabs/zord'
 
 import { nftPageHero } from './NFTPage.css'
 
-export interface NFTPageHeroProps extends BoxProps {}
+export interface NFTPageHeroProps extends BoxProps {
+  collectionAddress: string
+  tokenId: string
+}
+
+export function NFTPageHero({
+  className,
+  collectionAddress,
+  tokenId,
+  ...props
+}: NFTPageHeroProps) {
+  const { token } = useToken({ tokenId, collectionAddress })
```

Comment:
Same here, can we use the SSR token as initialData somehow rather than fetching?